rworr/pulley-shakesearch
# [Shakesearch] Fix tests via case-insensitive search, page sizes, page querying

I'd appreciate any feedback you have, in particular on style/best practices for Go/JS - especially JS

### Summary of changes:

Fixing `TestSearchCaseSensitive`
- Use regex + `FindAllIndex` to do a case-insensitive search
- Use `QuoteMeta` to make sure we wouldn't pass through any regex special chars

Fixing `TestSearchDrunk`
- Add consts for results per page, width of result
- only request first 20 matches from `FindAllIndex`

Fixing `should load more results for "horse" when clicking "Load More"`
- Add local page var (this feels like a weird way to do local state management, but I couldn't find anything better googling)
- append results to html table for subsequent pages (adding the str directly feels weird, LMK if there's a better way)
- add page query and use results to page through `FindAllIndex` results

Additional changes added (QoL / having fun with it):
- Add a "No more results!" output if page has gone past end of results
- Add a cache to hold all indexed results for a query to speed up `Load More` cases
- Improve error handling around pages and end-of-results - using `ebook` query in tests to confirm, since `ebook` appears at start and end of `completeworks.txt`
- Test helper functions
